File: wordNet/src/main/java/org/sqlunet/wordnet/sql/SynsetWordsQuery.java
/*
 * Copyright (c) 2019. Bernard Bou <1313ou@gmail.com>.
 */

package org.sqlunet.wordnet.sql;

import android.database.sqlite.SQLiteDatabase;

import org.sqlunet.sql.DBQuery;

/**
 * Query for words in synset
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
class SynsetWordsQuery extends DBQuery
{
	/**
	 * <code>QUERY</code> is the SQL statement
	 */
	static private final String QUERY = SqLiteDialect.SynsetWordsQuery;

	/**
	 * Constructor
	 *
	 * @param connection connection
	 * @param synsetId   target synset id
	 */
	@SuppressWarnings("boxing")
	public SynsetWordsQuery(final SQLiteDatabase connection, final long synsetId)
	{
		super(connection, SynsetWordsQuery.QUERY);
		setParams(synsetId);
	}

	/**
	 * Get word in result set
	 *
	 * @return word in the result set
	 */
	public String getWord()
	{
		assert this.cursor != null;
		return this.cursor.getString(0);
	}

	/**
	 * Get database id
	 *
	 * @return database id in the result set
	 */
	public long getId()
	{
		assert this.cursor != null;
		return this.cursor.getLong(1);
	}
}

File: wordNet/src/main/java/org/sqlunet/wordnet/sql/SynsetWordsQuery.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.wordnet.sql

import android.database.sqlite.SQLiteDatabase
import org.sqlunet.sql.DBQuery

/**
 * Query for words in synset
 *
 * @param connection connection
 * @param synsetId   target synset id
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
internal class SynsetWordsQuery(connection: SQLiteDatabase, synsetId: Long) : DBQuery(connection, QUERY) {

    init {
        setParams(synsetId)
    }

    /**
     * Get Word in the result set
     */
    val word: String
        get() = cursor!!.getString(0)

    /**
     * Database id in the result set
     */
    val id: Long
        get() = cursor!!.getLong(1)

    companion object {

        /**
         * `QUERY` is the SQL statement
         */
        private const val QUERY = SqLiteDialect.SynsetWordsQuery
    }
}